### ISSUE TITLE
Various runtimes

### ISSUE THREAD

**Issue by george99g** [yogstation13/yogstation-classic#109]
runtime error: Cannot execute null.reaction mob().
proc name: reaction mob (/datum/reagent/toxin/mutagen/reaction_mob)
  source file: Toxin-Reagents.dm,45
  usr: Llama Tosh (/mob/living/carbon/human)
  src: null
  usr.loc: the floor (158,109,1) (/turf/simulated/floor/plasteel)
  call stack:
reaction mob(Irene Hooker (/mob/living/carbon/human), 2, 5, 1, 0)
/datum/reagents (/datum/reagents): reaction(Irene Hooker (/mob/living/carbon/human), 2, 1, 1)
the beaker (/obj/item/weapon/reagent_containers/glass/beaker): attack(Irene Hooker (/mob/living/carbon/human), Llama Tosh (/mob/living/carbon/human), null)
Irene Hooker (/mob/living/carbon/human): attackby(the beaker (/obj/item/weapon/reagent_containers/glass/beaker), Llama Tosh (/mob/living/carbon/human), "icon-x=17;icon-y=18;left=1;scr...")
Irene Hooker (/mob/living/carbon/human): attackby(the beaker (/obj/item/weapon/reagent_containers/glass/beaker), Llama Tosh (/mob/living/carbon/human), "icon-x=17;icon-y=18;left=1;scr...")
Llama Tosh (/mob/living/carbon/human): ClickOn(Irene Hooker (/mob/living/carbon/human), "icon-x=17;icon-y=18;left=1;scr...")
Irene Hooker (/mob/living/carbon/human): Click(the floor (159,108,1) (/turf/simulated/floor/plasteel), "mapwindow.map", "icon-x=17;icon-y=18;left=1;scr...")
runtime error: Cannot read null.volume
proc name: reaction mob (/datum/reagent/proc/reaction_mob)
  source file: Chemistry-Reagents.dm,46
  usr: Jack Russo (/mob/living/carbon/human)
  src: null
  usr.loc: the floor (121,107,1) (/turf/simulated/floor/plasteel)
  call stack:
reaction mob(Jack Russo (/mob/living/carbon/human), 3, 1.66667, 1, 0.128571)
Water (/datum/reagent/water): reaction mob(Jack Russo (/mob/living/carbon/human), 3, 1.66667, 1, 0.128571)
/datum/reagents (/datum/reagents): reaction(Jack Russo (/mob/living/carbon/human), 3, 1, 1)
the space cleaner (/obj/item/weapon/reagent_containers/spray/cleaner): spray(the reinforced wall (121,104,1) (/turf/simulated/wall/r_wall))
runtime error: Cannot read null.volume
proc name: reaction mob (/datum/reagent/proc/reaction_mob)
  source file: Chemistry-Reagents.dm,46
runtime error: wrong type of value for list
proc name: updateSeeStaticMobs (/mob/living/simple_animal/drone/proc/updateSeeStaticMobs)
  source file: visuals_icons.dm,163
runtime error: wrong type of value for list
proc name: updateSeeStaticMobs (/mob/living/simple_animal/drone/proc/updateSeeStaticMobs)
  source file: visuals_icons.dm,163
runtime error: Cannot read null.volume
proc name: reaction mob (/datum/reagent/proc/reaction_mob)
  source file: Chemistry-Reagents.dm,46

**Comment by Cruix** [comment]
The issue with the first three and the last runtime is this comment in chemistry-reagents.dm:

//The reaction procs must ALWAYS set src = null, this detaches the proc from the object (the reagent)
//so that it can continue working when the reagent is deleted while the proc is still active.

All of the reagent procs that are runtiming are following this advice. I'm fairly certain that whoever wrote this does not know how setting src works. The only time I know of that setting src = null is useful is when the proc uses spawn() or sleep(), and none of the procs I have found in chemistry reagents do this. I could have missesd something, however. Even if I did not, it is terrible form to set src to null and not restore it immediately afterward, because with src set to null you cannot implicitly call any sibling procs, access any local instance variables, or use the ..() command.

I have tested removing all instances of "src = null" in the reagent code, and it seems to fix the problem. I need more coders' input before I make a pull request.
